mayastor/src/subsys/opts.rs

```diff
@@ -0,0 +1,458 @@
+            high_priority_weight: o.high_priority_weight,
+            nvme_adminq_poll_period_us: o.nvme_adminq_poll_period_us,
+            nvme_ioq_poll_period_us: o.nvme_ioq_poll_period_us,
+            io_queue_requests: o.io_queue_requests,
+            delay_cmd_submit: o.delay_cmd_submit,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct BdevOpts {
+    /// number of bdev IO structures in the shared mempool
+    bdev_io_pool_size: u32,
+    /// number of bdev IO structures cached per thread
+    bdev_io_cache_size: u32,
+}
+
+impl GetOpts for BdevOpts {
+    fn get(&self) -> Self {
+        let opts = spdk_bdev_opts::default();
+        unsafe {
+            spdk_sys::spdk_bdev_get_opts(
+                &opts as *const _ as *mut spdk_bdev_opts,
+            )
+        };
+        opts.into()
+    }
+
+    fn set(&self) -> bool {
+        let opts = Box::new(self.into());
+        if unsafe { spdk_bdev_set_opts(Box::into_raw(opts)) } != 0 {
+            return false;
+        }
+        true
+    }
+}
+
+impl Default for BdevOpts {
+    fn default() -> Self {
+        Self {
+            bdev_io_pool_size: 65535,
+            bdev_io_cache_size: 512,
+        }
+    }
+}
+
+impl From<spdk_bdev_opts> for BdevOpts {
+    fn from(o: spdk_bdev_opts) -> Self {
+        Self {
+            bdev_io_pool_size: o.bdev_io_pool_size,
+            bdev_io_cache_size: o.bdev_io_cache_size,
+        }
+    }
+}
+
+impl From<&BdevOpts> for spdk_bdev_opts {
+    fn from(o: &BdevOpts) -> Self {
+        Self {
+            bdev_io_pool_size: o.bdev_io_pool_size,
+            bdev_io_cache_size: o.bdev_io_cache_size,
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(default, deny_unknown_fields)]
+pub struct IscsiTgtOpts {
+    authfile: String,
+    /// none iqn name
+    nodebase: String,
+    /// timeout in seconds
+    timeout: i32,
+    /// nop interval in seconds
+    nop_ininterval: i32,
+    /// chap enabled
+    disable_chap: bool,
+    /// chap is required
+    require_chap: bool,
+    /// mutual chap
+    mutual_chap: bool,
+    /// chap group
+    chap_group: i32,
+    /// max number of sessions in th host
+    max_sessions: u32,
+    /// max connections per session
+    max_connections_per_session: u32,
+    /// max connections
+    max_connections: u32,
+    /// max queue depth
+    max_queue_depth: u32,
+    /// default
+    default_time2wait: u32,
+    /// todo
+    default_time2retain: u32,
+    /// todo
+    first_burst_length: u32,
+    ///todo
+    immediate_data: bool,
+    /// todo
+    error_recovery_level: u32,
+    /// todo
+    allow_duplicate_isid: bool,
+}
+
+impl Default for IscsiTgtOpts {
+    fn default() -> Self {
+        Self {
+            authfile: "".to_string(),
+            nodebase: "iqn.2019-05.io.openebs".to_string(),
+            timeout: 5,
```

Comment:
I wonder, if all children have the same timeout, would we want to make the frontend timeout the (child timeout x number of child io retries) + slack?

Reply:
The IO's are  sent out parallel, so I don't think we want to do that.

Reply:
not the retry IOs right? so for, if the frontend timeout is 1s, the child timeout is 1s with 5 retries, if even the first retry would not be useful because the frontend has already timeout? So the frontend IO timeout should allow for the retries plus some slack

Reply:
That assumes we do implicit retries without the initiator asking us to do so, is that what you mean?  We can talk about it some in the future what is best, but I think we dont want to do to many things on behalf of  the initiator here... :thinking:

Reply:
ah I thought that was the case now as I saw the retries being set to 5 on NvmeBdevOpts...
yes this is not the place to do it, I was just wondering :)